pkg/storage/segments/store.go

```diff
@@ -317,16 +317,16 @@ func (s *segmentStore) Repair(ctx context.Context, path storj.Path, lostPieces [
 	for j, v := range originalNodes {
 		if v != nil {
 			excludeNodeIDs = append(excludeNodeIDs, node.IDFromString(v.GetId()))
+			// remove all lost pieces from the list to have only healthy pieces
+			for i := range lostPieces {
+				if j == int(lostPieces[i]) {
+					v = nil
```

Comment:
`v` is a copy of the pointer stored in `originalNodes[j]`. Setting `v` to `nil` won't change `originalNodes[j]`. Check this example: https://play.golang.org/p/bkUa-5ZKKa3

Reply:
Addressed. Thoughts?

Reply:
This would work. I added a few more comments.